src/main/scala/com/gu/certificate/Magic.scala

```diff
@@ -109,11 +111,11 @@ object Magic extends BouncyCastle with FileHelpers {
 
   private def safeDomainString(domain: String) = domain.replace("*", "star")
 
-  private def assumeRole[T](roleArn: String)(block: AWSCredentialsProvider => T): T = {
+  private def assumeRole[T](underlyingProvider: AWSCredentialsProvider, roleArn: String)(block: AWSCredentialsProvider => T): T = {
```

Comment:
I don't think this function is needed anymore? I can pull it out later or we can take it out in this PR?